Ginger/Ginger/Run/RunSetsExecutionsHistoryPage.xaml.cs

```diff
@@ -728,29 +782,53 @@
 
         private void ExportUseCaseFromBusinessFlowExecutionSequence(BusinessFlowExecutionSequence bfSequence)
         {
-            BusinessFlow bf = GetBusinessFlowByName(bfSequence.Name);
-            IEnumerable<ActivitiesGroup> groups = bf
+            bool allActivityExistInSR = bfSequence
+                .ExecutedActivities
+                .All(executedActivity => executedActivity.ExistInSharedRepository);
+
+            if (!allActivityExistInSR)
+            {
+                throw new Exception($"All activities must be added to shared repository for generating use case.");
+            }
```

Comment:
All exceptions are handled in the parent method in the stack trace and shows detailed method to the user. Creating too many exception classes will bloat the solution since these are very niche edge cases.